crates/agent-sql/src/linked_materializations.rs

```diff
@@ -0,0 +1,91 @@
+use crate::{CatalogType, Id, TextJson};
+use proto_flow::flow::CaptureSpec;
+use serde_json::value::RawValue;
+
+pub struct Row {
+    pub materialization_name: String,
+    pub materialization_spec: TextJson<Box<RawValue>>,
+    pub last_pub_id: Id,
+}
+
+/// Returns all the materialization specs that have a `sourceCapture` matching one of the provided `capture_names`.
+pub async fn get_linked_materializations(
+    txn: &mut sqlx::Transaction<'_, sqlx::Postgres>,
+    capture_names: Vec<String>,
+) -> sqlx::Result<Vec<Row>> {
+    sqlx::query_as!(
+        Row,
+        r#"
+        select
+          catalog_name as materialization_name,
+          spec as "materialization_spec!: TextJson<Box<RawValue>>",
+          last_pub_id as "last_pub_id!: Id"
+        from live_specs
+        where spec_type = 'materialization' and spec->>'sourceCapture' = ANY ($1::catalog_name[])
```

Comment:
> what happens a materialization names a source capture that it's not authorized to read?

we check that when validating an upsert of the materialization.

> What happens if the grant existed, but has since been revoked?

we will continue to evolve its bindings, though we reserve the right to change that behavior in the future. However, the evolved publication will fail because its not authorized to the collections.

:+1:

Reply:
Any materialization in `live_specs` must have at one time been authorized to the `sourceCapture`. If the grant was since revoked, then the publication of the materialization will fail.

FWIW, I did add an index on `spec->>'sourceCapture'`. It's basic, and we may want to tweak it to exclude nulls or use `GIN`. But it seemed like a more reasonable starting point than full table scans. Do you think it'd be better to remove it for now?